taskfiles/utils.yml

```
@@ -141,3 +146,194 @@ tasks:
         \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
         -print0 | \
           xargs -0 clang-tidy {{.FLAGS}}
+
+  # ===
+  # CMAKE UTILS
+  # ===
+
+  # Runs cmake configure and build steps for the given source and build directories.
+  #
+  # @param {string} BUILD_DIR Cmake build directory to create.
+  # @param {string} SOURCE_DIR Project source directory containing the CMakeLists.txt file.
+  # @param {string={{.BUILD_DIR}}.md5} [CHECKSUM_FILE] Path to store the checksum of built files.
+  # @param {string=""} [CMAKE_ARGS] Any additional arguments to pass to cmake configure.
+  cmake-build:
+    label: "{{.TASK}}-{{.SOURCE_DIR}}-{{.BUILD_DIR}}"
+    internal: true
+    vars:
+      CHECKSUM_FILE: >-
+        {{default (printf "%s.md5" .BUILD_DIR) .CHECKSUM_FILE}}
+      CMAKE_ARGS: >-
+        {{default "" .CMAKE_ARGS}}
+    requires:
+      vars: ["BUILD_DIR", "SOURCE_DIR"]
+    sources:
+      - "{{.SOURCE_DIR}}/**/*"
+      - exclude: "{{.SOURCE_DIR}}/.cache/**/*"
+      - exclude: "{{.SOURCE_DIR}}/compile_commands.json"
+    generates: ["{{.CHECKSUM_FILE}}"]
+    deps:
+      - task: "validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_PATHS: ["{{.BUILD_DIR}}"]
+          EXCLUDE_PATHS: ["install_manifest.txt"]
+    cmds:
+      - >-
+        cmake
+        -S "{{.SOURCE_DIR}}"
+        -B "{{.BUILD_DIR}}"
+        {{.CMAKE_ARGS}}
+      - >-
+        cmake
+        --build "{{.BUILD_DIR}}"
+        --parallel
+      # This command must be last
+      - task: "compute-checksum"
+        vars:
+          DATA_PATHS: ["{{.BUILD_DIR}}"]
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+          EXCLUDE_PATHS: ["install_manifest.txt"]
+
+  # Runs cmake install step for the given build directory.
+  #
+  # @param {string} BUILD_DIR Cmake build directory.
+  # @param {string} INSTALL_PREFIX Path prefix for installing the project.
+  # @param {string={{.INSTALL_PREFIX}}.md5} [CHECKSUM_FILE] Path to store the checksum of installed
+  # files.
+  # @param {string[]=[{{.INSTALL_PREFIX}}]} [DATA_PATHS] Paths to compute the the checksum for.
+  cmake-install:
+    label: "{{.TASK}}-{{.BUILD_DIR}}-{{.INSTALL_PREFIX}}"
+    internal: true
+    vars:
+      CHECKSUM_FILE: >-
+        {{default (printf "%s.md5" .INSTALL_PREFIX) .CHECKSUM_FILE}}
+      # Convert the install prefix to a single element array in the default case where DATA_PATHS
+      # has not been set.
+      _INSTALL_PREFIX_ARRAY: ["{{.INSTALL_PREFIX}}"]
+      DATA_PATHS:
+        ref: "default ._INSTALL_PREFIX_ARRAY .DATA_PATHS"
```

```suggestion
      DATA_PATHS:
        ref: "default (list .INSTALL_PREFIX) .DATA_PATHS"
```

I think that's equivalent?